animations/lib/src/misc/focus_image.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter/material.dart';
+
+class FocusImageDemo extends StatelessWidget {
+  static String routeName = '/misc/focus_image';
+
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('Focus Image')),
+      body: Grid(),
+    );
+  }
+}
+
+class Grid extends StatelessWidget {
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: GridView.builder(
+        itemCount: 40,
+        gridDelegate:
+        SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 4),
+        itemBuilder: (context, index) {
+          return (index >= 20)
+              ? SmallCard('assets/eat_cape_town_sm.jpg')
+              : SmallCard('assets/eat_new_orleans_sm.jpg');
+        },
+      ),
+    );
+  }
+}
+
+Route _expandToPageRoute(BuildContext parentContext, String image) {
+  Widget transitionsBuilder(
+      BuildContext context,
+      Animation<double> animation,
+      Animation<double> secondaryAnimation,
+      Widget child,
+      ) {
+    var rectAnimation = _rectTween(parentContext)
+        .chain(CurveTween(curve: Curves.ease))
+        .animate(animation);
+
+    return Stack(
+      children: [
+        PositionedTransition(rect: rectAnimation, child: child),
+      ],
+    );
+  }
+
+  return PageRouteBuilder(
+    pageBuilder: (context, animation, secondaryAnimation) {
+      return _SecondPage(image);
+    },
+    transitionsBuilder: transitionsBuilder,
+  );
+}
+
+Tween<RelativeRect> _rectTween(BuildContext context) {
```

Comment:
nit: Try a name that clearly indicates the function is creating a tween ("_createRectTween" or similar).